types/types.go

```diff
@@ -0,0 +1,16 @@
+package types
+
+import (
+	"context"
+)
+
+//go:generate mockgen -package mocks -destination mocks/challenge_verifier.go . ChallengeVerifier
+
+type ChallengeVerificationResult struct {
+	Hash   []byte
+	NodeId []byte
+}
+
+type ChallengeVerifier interface {
+	Verify(ctx context.Context, challenge, signature []byte) (*ChallengeVerificationResult, error)
+}
```

Comment:
I'm generally against `types` packages. Imo they shouldn't exist. Either the interface should be at the users side, or at the implementers side, depending on who has the authority over it.

With that I mean: why does this interface exist?
- So it can be mocked? -> Then it should be at the users side.
- So it can be used for hiding implementation specifics behind a general interface (what imo is happening here)? Then it should be part of the implementation.

Reply:
Moved the interface and its errors to `gateway/challenge_verifier`.